### DISCUSSION TITLE
What happens if you instantiate a Db entity but don't append it?

### DISCUSSION THREAD

**Discussion by kwollaston** [CEXT-Dan/PyRx#80] [Q&A] [answered]
I have written a function that creates a Db.MText entity in order to determine the width of text so I can scale the table's column size based on the longest value. It does work but I'm wondering if there are any side effects.

If I don't run model.appendAcDBEntity(mtext), what happens? Does this get picked up by GC? Is it still hidden in the database? I don't want the text to appear in the model but also don't want to fill up the database without knowing.

**Comment by CEXT-Dan** [comment, marked as answer]
Garbage collected.

Basically, if the object has an objectid, it’s closed, else deleted 
https://github.com/CEXT-Dan/PyRx/blob/751da54f49eeb62a2b450b4bc28ce4029862204f/PyRxCore/PyRxObject.cpp#L58-L72
There’s a couple of rare conditions when the object is handed over to AutoCAD
